content/dev/design-pattern-abstract-factory.md

```diff
@@ -0,0 +1,173 @@
+description:    "Deuxième article d'une série consacrée aux Design Patterns. Aujourd'hui : le pattern Abstract Factory"
+
+thumbnail:      "/images/posts/design-pattern/creation-abstract-factory.png"
+header_img:     "/images/posts/design-pattern/creation-abstract-factory.png"
+tags:           ["Design Pattern"]
+categories:     ["Dev", "Design Pattern"]
+
+author_username:    "xavierr"
+
+---
+
+Deuxième article d'une série consacrée aux Design Patterns. Aujourd'hui, le pattern `AbstractFactory` où il sera question de produits, de familles et de fabriques (_factories_, au pluriel s'il vous plaît).
+
+<table width="100%">
+    <tr>
+        <td width="50%" align="center">
+            {{< figure style="text-align:center;" src="/images/posts/design-pattern/img/ingals-family.jpeg" alt="Une gentille famille américaine">}}
+    <figcaption style="text-align: center;font-style: italic;">Une gentille famille américaine</figcaption>
+        </td>
+        <td width="50%" align="center">
+            {{< figure src="/images/posts/design-pattern/img/chainsaw-family.jpeg" alt="Une famille américaine">}}
+<figcaption style="text-align: center;font-style: italic;">Une <strike>gentille</strike> famille américaine</figcaption>
+        </td>
+    </tr>
+</table>
+
+<div style="border-left: 5px solid #ffa600;padding: 20px;margin: 20px 0;">
+    Le livre <strong><i>Head First Design Patterns</i></strong> (<a href="/fr/dev/design-pattern-factory-method">dont j'ai déjà vanté les mérites</a>) regroupe les deux patterns <code>Factory Method</code> et <code>AbstractFactory</code> dans un même chapitre consultable en ligne et intitulé <a href="https://www.safaribooksonline.com/library/view/head-first-design/0596007124/ch04.html" target="_blank">The Factory Pattern: Baking with OO Goodness</a>. Je ne saurais trop vous encourager à le consulter !
+</div>
+
+## Classification
+
+Le pattern `Abstract Factory` est classé dans la catégorie des __Design Patterns de création__.
+
+## Définition
+
+> Provide an interface for creating families of related or dependent objects without specifying their concrete classes.
+
+C'est une très jolie définition dont je ne me lasse pas ... (Comme toute définition, elle paraît assez barbare tant que l'on n'a pas approfondi le concept, ce que je vous propose de faire dans les paragraphes qui suivent).
+
+En résumé, le pattern `Abstract Factory` va nous permettre d'instancier des familles de produits dépendant les uns des autres sans qu'il soit nécessaire de préciser leur type concret (je ne suis pas sûr qu'on soit plus avancé ...)
+
+## Schéma du design pattern `Abstract Factory`
+
+<p class="text-center">
+    {{< figure class="text-center" src="/images/posts/design-pattern/creation-abstract-factory.png" alt="Le Design Pattern 'Abstract Factory'">}}
+</p>
+
+Ne vous laissez pas impressionner par la densité du schéma et le nombre de participants. Le pattern n'a rien d'insurmontable et peut s'avérer utile dans de nombreuses situations.
```

Comment:
> Ne vous laissez pas impressionner par la densité du schéma et le nombre de participants

![hmhm](https://img.memesuper.com/d2dc3df0a2c00a2835b50921ce9322ec_if-you-know-what-i-mean-mr-mr-bean-meme-if-you-know-what-i-mean_400-400.jpeg)